pkg/nirmata/aks_clusterType.go

```diff
@@ -212,6 +215,17 @@ func resourceClusterTypeCreate(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
+	clusterTypeID, err := apiClient.QueryByName(client.ServiceClusters, "clustertypes", name)
+	if err != nil {
+		fmt.Printf("Error ", err)
+		return err
+	}
+
+	err = apiClient.WaitForState(clusterTypeID,"state","ready",d.Timeout(schema.TimeoutCreate),"Failed to get cluster status")
```

Comment:
I don't think this logic is correct. The `waitForState` will wait for `ready` state. But what happens if the state goes to `failed` or some other error state?